### ISSUE TITLE
Update examples

### ISSUE THREAD

**Issue by iamcharleschege** [rustasync/runtime#13]
This project looks promising. I like the simplicity. However, can you kindly update your examples with every release to prevent errors. I have tried a few and some of them result in errors, for example, the UDP and TCP echo server examples result in error `error[E0599]: no method named `next` found for type `runtime::net::tcp::Incoming<'_>` in the current scope
  --> src/main.rs:18:46 `

**Comment by iamcharleschege** [comment]
The echo server in the description page also shows nothing when using `nc -u localhost 8080`

**Comment by yoshuawuyts** [comment]
@iamcharleschege for the UDP Echo example, after you start typing things and hit enter it'll respond. I just tried it standalone, and it seems to work!

![2019-04-23-125951_3840x2160](https://user-images.githubusercontent.com/2467194/56576050-aacd7680-65c7-11e9-88ea-ffd3c449fe7b.png)

**Comment by yoshuawuyts** [comment]
Also confirmed the TCP Echo example works outside of the repo.

![2019-04-23-130311_3840x2160](https://user-images.githubusercontent.com/2467194/56576220-131c5800-65c8-11e9-8c2d-c108e69f92a5.png)

---

I'm not sure what's going on. I've uploaded a repro of the TCP Example here: https://github.com/yoshuawuyts/repro-runtime-13/blob/master/src/main.rs. Everything seems to be working as expected.

The error you're seeing would imply that `futures::prelude::*` is somehow not importing the `Stream` trait. I'm not sure why that might be occurring, but perhaps if you share more of your platform details I could help figure out what's going on?

**Comment by ZhouHansen** [comment]
Encounter the same error, and add more details：

```sh
error[E0599]: no method named `next` found for type `runtime::net::tcp::Incoming<'_>` in the current scope
  --> src/main.rs:18:46
   |
18 |     while let Some(stream) = await!(incoming.next()) {
   |                                              ^^^^
   |
   = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope, perhaps add a `use` for it:
   |
8  | use futures_util::stream::StreamExt;
   |

```

Platform: 
macOS 10.13.6
node v11.0.0
npm 6.8.0

cargo.toml:
```toml
[dependencies]
runtime = "0.3.0-alpha.1"
runtime-native = "0.3.0-alpha.1"
runtime-raw = "0.3.0-alpha.1"
futures = "0.1.26"
rand = "0.6.5"
failure = "0.1.5"
```

**Comment by iamcharleschege** [comment]
The example https://github.com/yoshuawuyts/repro-runtime-13/blob/master/src/main.rs gives the same error

**Comment by iamcharleschege** [comment]
Environment is Debian 9,
 Rustc version is `Rustc 1.35.0-nightly (aa99abeb2 2019-04-14)`

`Cargo.toml ` file

```
edition = "2018"

[dependencies]
runtime = "0.3.0-alpha.2"
futures = "0.1.26"
```

**Comment by yoshuawuyts** [comment]
@ZhouHansen @iamcharleschege Ah! -- could you try using the following in your Cargo.toml:
```toml
edition = "2018"

[dependencies]
runtime = "0.3.0-alpha.2"
futures-preview = "0.3.0-alpha.14"
```

Right now the `futures` crate still builds for futures 0.1, whereas `futures-preview` uses futures 0.3. Both can be used in code as `use futures;` without a problem.

This feels like something we should perhaps mention in our documentation too. Sorry about this!

**Comment by ZhouHansen** [comment]
It works! and maybe we should mention the dependencies for each example.

[Hello World](https://github.com/rustasync/runtime/blob/master/examples/hello.rs) example requires 
```toml
[dependencies]
runtime-native = "0.3.0-alpha.1"
runtime-raw = "0.3.0-alpha.1"
```

**Comment by iamcharleschege** [comment]
This works! Please add to documentation

**Comment by yoshuawuyts** [comment]
@ZhouHansen the dependency problems should be fixed now in `runtime@0.3.0-alpha.2`; so all you need is `runtime`.

We had some problems with our macros that we've since fixed (:

**Comment by yoshuawuyts** [comment]
#16 should fix this going forward; happy it now all works. Thanks for the feedback! going to close this for now :sparkles:
